Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/ContactDiaryPreferences.kt

```diff
@@ -0,0 +1,27 @@
+package de.rki.coronawarnapp.contactdiary.ui
+
+import android.content.Context
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import de.rki.coronawarnapp.util.preferences.createFlowPreference
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ContactDiaryPreferences @Inject constructor(
```

Comment:
Can you please elaborate on the difference between `ContactDiaryPreferences ` and `ContactDiarySettings `

Reply:
Divided things that don't necessarily belong together. ContactDiaryPreferences manages android preferences for the contact diary part of the app (persistence layer). ContactDiarySettings is one layer above (domain layer) and is supposed to manage the settings for the contact diary part of the app from a functional perspective. Persisting the settings is just one way to do it but could be changed. By dividing these two layers i was able to test the settings by mocking the persistence layer.

Reply:
thx.